tests/message-extensions/src/index.ts

```diff
@@ -128,6 +130,70 @@ app.on("message.ext.query", async ({ activity }) => {
 });
 // :snippet-end: message-ext-query
 
+// :snippet-start: message-ext-select-item
+app.on("message.ext.select-item", async ({ activity, send }) => {
+  const { option } = activity.value;
+
+  await send(`Selected item: ${option}`);
+
+  return {
+    status: 200,
+  }
+});
+// :snippet-end: message-ext-select-item
+
+// :snippet-start: message-ext-query-settings-url
+app.on("message.ext.query-settings-url", async ({ activity }) => {
+  // Get user settings from storage if available
+  const userSettings = await app.storage.get(activity.from.id) || { selectedOption: '' }
+  const escapedSelectedOption = encodeURIComponent(userSettings.selectedOption);
+
+  return {
+    composeExtension: {
+      type: "config",
+      suggestedActions: {
+        actions: [
+          {
+            type: "openUrl",
+            title: "Settings",
+            // ensure the bot endpoint is set in the environment variables
+            // process.env.BOT_ENDPOINT is not populated by default in the Teams Toolkit setup. 
+            value: `${process.env.BOT_ENDPOINT}/settings?selectedOption=${escapedSelectedOption}`
+          }
+        ]
+      }
+    }
+  };
+});
+// :snippet-end: message-ext-query-settings-url
+
+// :snippet-start: message-ext-setting
+app.on("message.ext.setting", async ({ activity, send }) => {
+  const { state } = activity.value
+  if (state == "CancelledByUser") {
+    return {
+      status: 400
+    }
+  }
+  const selectedOption = state;
+  
+  // Save the selected option to storage
+  await app.storage.set(activity.from.id, { selectedOption })
+  
+  await send(`Selected option: ${selectedOption}`)
+
+  return {
+    status: 200
+  }
+});
+// :snippet-end: message-ext-setting
+
+// :snippet-start: message-ext-serve-html
+app.http.use(`/settings`, async (_, res) => {
```

Comment:
use app.tab. See how did it in another sample

Reply:
good idea. updated